src/lib/transaction_snark/test/zkapp_tokens/zkapp_tokens.ml

```diff
@@ -0,0 +1,186 @@
+                    in
+                    nonce
+                  in
+                  let create_token_parties =
+                    let nonce = nonce_from_ledger () in
+                    let with_dummy_signatures =
+                      Parties_builder.mk_forest
+                        [ Parties_builder.mk_node
+                            (Parties_builder.mk_party_body Call token_funder
+                               Token_id.default
+                               (-(4 * account_creation_fee)) )
+                            []
+                        ; Parties_builder.mk_node
+                            (Parties_builder.mk_party_body Call token_owner
+                               Token_id.default (3 * account_creation_fee) )
+                            []
+                        ]
+                      |> Parties_builder.mk_parties_transaction ~fee:7
+                           ~fee_payer_pk:pk ~fee_payer_nonce:nonce
+                    in
+                    Parties_builder.replace_authorizations ~keymap
+                      with_dummy_signatures
+                  in
+                  let%bind () =
+                    U.check_parties_with_merges_exn ledger
+                      [ create_token_parties ]
+                  in
+                  ignore
+                    ( ledger_get_exn ledger
+                        (Public_key.compress token_owner.public_key)
+                        Token_id.default
+                      : Account.t ) ;
+                  let mint_token_parties =
+                    let nonce = nonce_from_ledger () in
+                    let with_dummy_signatures =
+                      Parties_builder.mk_forest
+                        [ Parties_builder.mk_node
+                            (Parties_builder.mk_party_body Call token_owner
+                               Token_id.default (-account_creation_fee) )
+                            [ Parties_builder.mk_node
+                                (Parties_builder.mk_party_body Call
+                                   token_account1 custom_token_id 100 )
+                                []
+                            ]
+                        ]
+                      |> Parties_builder.mk_parties_transaction ~fee:7
+                           ~fee_payer_pk:pk ~fee_payer_nonce:nonce
+                    in
+                    Parties_builder.replace_authorizations ~keymap
+                      with_dummy_signatures
+                  in
+                  let%bind () =
+                    U.check_parties_with_merges_exn ledger
+                      [ mint_token_parties ]
+                  in
+                  check_token_balance token_account1 100 ;
+                  let token_transfer_parties =
+                    let nonce = nonce_from_ledger () in
+                    let with_dummy_signatures =
+                      Parties_builder.mk_forest
+                        [ Parties_builder.mk_node
```

Comment:
Could you add these as well? (these could be in a different PR)
1) transfers using another new token id in the same transaction
2) transfers involving more than 1 custom token ids within the same tree
3) sprinkle some mina transfers in a tree with custom token transfers

Reply:
Will add in a new PR.